data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -0,0 +1,55 @@
+package com.amazon.dataprepper.model.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+/**
+ * Model class for a Pipeline containing all possible Plugin types in Configuration YAML
+ *
+ * @since 1.2
+ */
+public class PipelineModel {
+
+    private final PluginModel sourcePluginModel;
+    private final List<PluginModel> prepperPluginModel;
+    private final List<PluginModel> sinkPluginModel;
+    private Integer workers = 1;
```

Comment:
@dlvenable I decided to remove the default values since I'm calling them in the constructor and updated it to:
```
    private final Integer workers;
    private final Integer readBatchDelay;
```
If they're null, then `null` values will be passed to the Constructor.